src/twisted/internet/test/test_fdset.py

```diff
@@ -115,6 +115,19 @@ def fail():
         # Getting here means the right thing happened probably.
 
 
+    def test_removeReader_selectables_nokey(self):
+        """
+        L{reactor.removeReader()} accepts an L{IReadDescriptor} provider
+        previously passed to C{reactor.addReader()}. Checks if it doesn't
+        raise KeyError exception
+        """
+        reactor, fd, server = self._simpleSetup()
+        reactor.addReader(fd)
+        reactor._selectables = {}
```

Comment:
Can you provoke this failure in a more realistic way?  It doesn't seem likely that the affected application goes around erasing the reactor's internal state.  A realistic test would provide more confidence that this really fixes the problem.  Also, it would embody a more complete understanding of the cause of the failure which will assist future maintainers of the reactor and test suite.

Reply:
I see this a lot with ldaptor running under Crochet, so might be a similar problem with crossing thread boundaries there.

Reply:
A minimal reproducing example would go a long way towards helping make progress on this.